airbyte-commons/src/main/java/io/airbyte/commons/json/JsonSchemas.java

```diff
@@ -189,8 +189,7 @@ private static void traverseJsonSchemaInternal(final JsonNode jsonSchemaNode,
             // hit every node.
             traverseJsonSchemaInternal(jsonSchemaNode.get(JSON_SCHEMA_ITEMS_KEY), newPath, consumer);
           } else {
-            throw new IllegalArgumentException(
-                "malformed JsonSchema array type, must have items field in " + jsonSchemaNode);
+            log.warn("The array is missing an items field. The traversal is silently stopped.");
```

Comment:
Are we saying this is a valid case in the end? If so, it should probably not be a warning. Also, I wonder if we should log the `jsonSchemaNode`, otherwise, it is not clear what we can leverage from this log.

Reply:
+1 to logging some additional context to help when attempting to debug when catalogs are hitting this state.

Reply:
I think the warning makes sense because we are actively choosing to ignore part of the old, malformed schema.  The warning will help us be notified that some connectors might have a bad SPEC that needs to be fixed.

Reply:
It is something that we want to get rid of on the long term. It is not clear what is generating it. I have added the jsonSchemaNode to the log

Reply:
Feels like we should add a way to track violations for connectors in order to help prioritize fixes. I suspect that this warning will be missed most of the times and probably add more confusion/noise when looking into an error.

Reply:
Adding to this, when debugging the current exception being thrown just using the jsonSchemaNode, it was tough to know where specifically this was happening. It would also be useful to print the path if possible. +1 on tracking these as well.